### PR TITLE
FAI-760 gov login integration

### PR DESCRIPTION
**Pull request by dashton82**

### CODE REVIEW

**Review by balaji-j — DISMISSED** [review]

**Review by balaji-j — APPROVED** [review]

File: src/SFA.DAS.AssessorService.Web/ViewModels/Account/ChangeSignInDetailsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace SFA.DAS.AssessorService.Web.ViewModels.Account;
+
+public class ChangeSignInDetailsViewModel
+{
+    private readonly string _integrationUrlPart = "";
+    public ChangeSignInDetailsViewModel(string environment)
+    {
+        if (!environment.Equals("prd", StringComparison.CurrentCultureIgnoreCase))
```
[the remark this answers is not in the document]
Comment (dashton82) [reply src/SFA.DAS.AssessorService.Web/ViewModels/Account/ChangeSignInDetailsViewModel.cs:10]: its the ResrouceEnvironmentName environment variable we are using to determine the URL to be used for going to the settings for gov login. There are only two URLs, one for prod and one for everything else - I'm not sure as to what value it adds putting it all into config - apart from having more config 😁

File: src/SFA.DAS.AssessorService.Application/Handlers/UserManagement/InviteContactToOrganisationHandler.cs [hunk 2]
```diff
@@ -43,16 +48,27 @@ public async Task<InviteContactToOrganisationResponse> Handle(InviteContactToOrg
...
             var organisation = await _organisationQueryRepository.Get(request.OrganisationId);
-            
+
             var newContact = await CreateNewContact(request, organisation);
 
-            var inviter = await _contactQueryRepository.GetContactById(request.InvitedByContactId);
-                
-            await _signInService.InviteUserToOrganisation(request.Email, request.GivenName, request.FamilyName, newContact.Id, organisation.EndPointAssessorName, inviter.DisplayName);
+            // send invite email to the user with service link.
+            if (_apiConfiguration.UseGovSignIn)
```
[the remark this answers is not in the document]
Comment (dashton82) [reply src/SFA.DAS.AssessorService.Application/Handlers/UserManagement/InviteContactToOrganisationHandler.cs:55]: These are emails that are currently sent through das-login, they only need to be sent when gov login is on

File: src/SFA.DAS.AssessorService.Web/Helpers/EnvironmentHelper.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+namespace SFA.DAS.AssessorService.Web.Helpers;
+
+public static class EnvironmentHelper
```
[the remark this answers is not in the document]
Comment (dashton82) [reply src/SFA.DAS.AssessorService.Web/Helpers/EnvironmentHelper.cs:3]: I can update it to use the configuration value, but i'd argue that reducing configuration should be the goal, this is currently testable and we know the value to be obtained, as soon as we pull from configuration we lose that validation

[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AssessorService.Web/Helpers/EnvironmentHelper.cs:3]: This should be using configuration you don't need to unit test the configuration

File: src/SFA.DAS.AssessorService.Web/ViewModels/Account/ChangeSignInDetailsViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AssessorService.Web/ViewModels/Account/ChangeSignInDetailsViewModel.cs:10]: You could try to use that argument to justify moving almost all configuration into the code base - apart from the secrets - I don't buy that argument, the code should have no knowledge of the names or values for specific environments it is deployed to

**Pull request merged**